### PR TITLE
Add code owner entry for the mobile API team

### PR DESCRIPTION
**Pull request by kreek**
## Description of change
Creates a code owner entry so that the [mobile API team](https://github.com/orgs/department-of-veterans-affairs/teams/mobile-api-team) owns code in the `modules/mobile` dir and associated spec support files. I copied the VAOS model where both the VAOS team and `backend-review-group` are code owners. Is this correct? We want the mobile team to own it and be able to approve for merge. Then also have the backend review group pinged for visibility, but not require their ✅ to merge.

### CODE REVIEW

**Review by jperk51 — APPROVED** [review]
Looks good to me

File: .github/CODEOWNERS
Comment (erluti) [inline .github/CODEOWNERS:16]: anything outside of `modules/mobile` is shared code that VSP has to support, so making mobile-api-team code owners outside that directory isn't going to work.  Moving code that doesn't need to be shared with the platform into `modules/mobile` would be a good approach to own some of that stuff.

**Review by erluti — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (erluti) [reply .github/CODEOWNERS]: Did you want to own `spec/support/*/mobile/`?  Originally you were going to copy vaos and thought that was a good approach.  But I'll approve it without that if you want

Comment (kreek) [reply .github/CODEOWNERS]: @erluti we were going to try to have our spec support files also in the engine (still researching how to do this). But ya for now I'll add the mobile support dir back in.

**Review by erluti — APPROVED** [review]

**Pull request merged**
